src/savegame_json.cpp

```diff
@@ -1560,13 +1559,18 @@ void vehicle_part::deserialize(JsonIn &jsin)
     data.read("target_second_y", target.second.y);
     data.read("target_second_z", target.second.z);
 
-    // migrate legacy base items which may not be tagged VEHICLE
-    if( !base.has_flag( "VEHICLE") ) {
-        base.item_tags.insert( "VEHICLE" );
-        if( base.is_magazine() ) {
-            base.ammo_set( id.obj().fuel_type, amount );
-            amount = 0;
+    // with VEHICLE tag migrate fuel tanks only if amount field exists
+    if( base.has_flag( "VEHICLE") ) {
+        if( data.has_int( "amount" ) && ammo_capacity() > 0 && id.obj().fuel_type != "battery" ) {
```

Comment:
The first migration set `VEHICLE` and migrated `battery` but still wrote out `amount` as a zero value.